File: mainActivity/src/main/java/com/sierrawireless/avphone/service/LogMessage.java
package com.sierrawireless.avphone.service;

import android.content.Intent;

public class LogMessage extends Intent {

    public static final String LOG_EVENT = "com.sierrawireless.avphone.event.log";

    // keys used for broadcasting log events
    private static final String LOG = "log";
    private static final String ALARM = "alarm";

    public LogMessage(String message, boolean alarm) {
        super(LOG_EVENT);

        this.putExtra(LOG, message);
        this.putExtra(ALARM, alarm);
    }

    public String getMessage() {
        return this.getStringExtra(LOG);
    }
    public boolean getAlarm() { return this.getBooleanExtra(ALARM, false);}

}

File: mainActivity/src/main/java/com/sierrawireless/avphone/service/LogMessage.kt
package com.sierrawireless.avphone.service

import android.content.Intent

class LogMessage(message: String, alarm: Boolean) : Intent(LOG_EVENT) {

    val message: String
        get() = this.getStringExtra(LOG)
    val alarm: Boolean
        get() = this.getBooleanExtra(ALARM, false)

    init {

        this.putExtra(LOG, message)
        this.putExtra(ALARM, alarm)
    }

    companion object {

        const val LOG_EVENT = "com.sierrawireless.avphone.event.log"

        // keys used for broadcasting log events
        const val LOG = "log"
        private const val ALARM = "alarm"
    }

}
